app/models/invoice.rb

```diff
@@ -263,6 +265,18 @@ def different_boundaries_for_subscription_and_charges(subscription)
     subscription_from != charges_from && subscription_to != charges_to
   end
 
+  def mark_as_disputed!(timestamp = Time.current)
+    self.payment_dispute_lost_at ||= timestamp
+    self.payment_disputed = true
```

Comment:
Is there a sceneario where you have `payment_disputed = true` but `payment_dispute_lost_at = nil?`? Can we have disputed payment that you don't lose or if we "win", we `mark_as_not_disputed!`?

I'm thinking the `payment_disputed` could be a timestamp `payment_disputed_at` instead of a boolean, like we do for `discarded_at` for instance. If the lost_at can be different, we can have 2 timestamps, other only keep `payment_disputed_at`.

I'm reading the spec now, sorry I didn't review it earlier 🙏